File: go/euler25_alternate.go
//
// projecteuler.net
//
// problem 25
//

package main

import (
	"fmt"
	"math"
)

func main() {
	log_phi := math.Log(Math.Phi)
	x := math.Log(10) * 999
	y := math.Log(5) / 2
	n := math.Ceil((x + y) / log_phi)
	fmt.Printf("%v\n", n)
}

File: python/euler25_alternate.py
#! /usr/bin/env python
# -*- coding: utf-8 -*-
# projecteuler.net
#
# problem 25
#

import math

log_phi = math.log(1.618033988749895)
x = math.log(10) * 999
y = math.log(5) / 2
n = math.ceil((x + y) / log_phi)

print(n)